lib/src/components/NxFileUpload/NxMultiFileUpload/stateful/NxStatefulMultiFileUpload.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { forwardRef, useState } from 'react';
+import NxMultiFileUpload from '../NxMultiFileUpload';
+import { StatefulProps as Props, statefulPropTypes } from '../../types';
+
+export { Props };
+
+const NxStatefulMultiFileUpload = forwardRef<HTMLDivElement, Props>(function NxStatefulMultiFileUpload(props, ref) {
+  const { onChange: onChangeProp, ...otherProps } = props,
+      [files, setFiles] = useState<FileList | null>(null),
+      [isPristine, setIsPristine] = useState(true);
+
+  function onChange(files: FileList | null) {
+    setIsPristine(false);
```

Comment:
Use the state helpers rather than doing this manually

Reply:
> Use the state helpers rather than doing this manually

Added in state helpers here: [5cfa80f](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/5cfa80f73e292c8252e1833e64caad1c14a580e7)